perception/autoware_tensorrt_yolox/src/tensorrt_yolox_node.cpp

```diff
@@ -109,58 +116,82 @@ TrtYoloXNode::TrtYoloXNode(const rclcpp::NodeOptions & node_options)
 
   objects_pub_ = this->create_publisher<tier4_perception_msgs::msg::DetectedObjectsWithFeature>(
     "~/out/objects", 1);
-  mask_pub_ = image_transport::create_publisher(this, "~/out/mask");
-  color_mask_pub_ = image_transport::create_publisher(this, "~/out/color_mask");
-  image_pub_ = image_transport::create_publisher(this, "~/out/image");
+  if (is_using_image_transport_) {
+    mask_pub_ = image_transport::create_publisher(this, "~/out/mask");
+    color_mask_pub_ = image_transport::create_publisher(this, "~/out/color_mask");
+    image_pub_ = image_transport::create_publisher(this, "~/out/image");
+  } else {
+    simple_image_pub_ = this->create_publisher<sensor_msgs::msg::Image>("~/out/image", 1);
+    simple_mask_pub_ = this->create_publisher<sensor_msgs::msg::Image>("~/out/mask", 1);
+    simple_color_mask_pub_ = this->create_publisher<sensor_msgs::msg::Image>("~/out/color_mask", 1);
+  }
 
   if (declare_parameter("build_only", false)) {
     RCLCPP_INFO(this->get_logger(), "TensorRT engine file is built and exit.");
     rclcpp::shutdown();
   }
 }
 
-void TrtYoloXNode::onConnect()
+bool TrtYoloXNode::checkInputBlocked()
 {
-  using std::placeholders::_1;
-  if (
-    objects_pub_->get_subscription_count() == 0 &&
-    objects_pub_->get_intra_process_subscription_count() == 0 &&
-    image_pub_.getNumSubscribers() == 0 && mask_pub_.getNumSubscribers() == 0 &&
-    color_mask_pub_.getNumSubscribers() == 0) {
-    image_sub_.shutdown();
-  } else if (!image_sub_) {
+  bool result;
+  if (type_adaptation_activated_) {
+    result = (gpu_image_sub_ == nullptr);
+  } else {
+    result = !image_sub_;
+  }
+  return result;
```

Comment:
[NITS] We can write simple.
```suggestion
  return type_adaptation_activated_ ? gpu_image_sub_ == nullptr : !image_sub_;
```